lib/helper.js

```diff
@@ -141,6 +141,10 @@ const replaceWinPath = (path) => {
 
 exports.normalizeWinPath = process.platform === 'win32' ? replaceWinPath : _.identity
 
+exports.getFileType = (file) => {
+  return file.type || path.extname(file.path.split(/[?#]/, 1)[0] || '').substring(1)
```

Comment:
Is there any specific reason why we need ` || ''` here? 
Otherwise LGTM.

Reply:
Also it may be better to first take extensions and then split as file name may contain `?` or `#` (as it was before). E.g. `myfile?.js` or `myfile#.js` are valid filenames on Linux.

Hm, actually `path.extname('http://example.com/mylib.js?load=test.md')` will fail result in `md`, not `js`. Maybe we need a more robust logic here...

Reply:
> Is there any specific reason why we need || '' here?

String#split method can return an empty array.

> Probably we need a more robust logic here...

Yes. I'm expecting that someone write a stricter implementation later.

Reply:
Hm, looks like the source code treats paths only as URLs.

Reply:
> Yes. I'm expecting that someone write a stricter implementation later.

Well, current implementation breaks file type detection for `myfile?.js` and `myfile#.js` in favour of URLs. I don't think it is appropriate to break one corner case in favour of another.

Here is a better idea. Instead of having single method to handle both file path and URL we move the file type detection to constructors in `file.js` and `url.js` correspondingly. This way we can provide appropriate implementation depending on what we're dealing with.

E.g. 

`path.extname(filePath).substring(1)` for file paths
`path.extname(new URL('https://example.com/myfile.js?test=param').pathname).substring(1)` for URLs

Reply:
Not sure. You should do such refactoring yourself.

Reply:
Okey, I'll try and submit an alternative PR in the coming days.